src/components/tooltip/tooltip.pw.tsx

```diff
@@ -0,0 +1,464 @@
+    const buttonElement = page.getByRole("button");
+    await buttonElement.hover();
+    await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
+  });
+
+  test(`should hide tooltip when mouse leaves target`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<UncontrolledTooltipComponent />);
+
+    await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+    const buttonElement = page.getByRole("button");
+    await buttonElement.hover();
+    await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
+    await page.mouse.move(100, 100);
+    await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+  });
+
+  test(`should show tooltip when target is focused`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<UncontrolledTooltipComponent />);
+
+    await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+    const buttonElement = page.getByRole("button");
+    await buttonElement.focus();
+    await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
+  });
+
+  test(`should hide tooltip when target is blurred`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<UncontrolledTooltipComponent />);
+
+    await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+    const buttonElement = page.getByRole("button");
+    await buttonElement.focus();
+    await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
+    await buttonElement.blur();
+    await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+  });
+
+  test(`new tooltip target should still trigger tooltip visibility`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TooltipWithChangingTargetComponent />);
+
+    const buttonElement = await getDataElementByValue(page, "main-text");
+    await expect(buttonElement).toHaveText("Target");
+    await buttonElement.hover();
+    await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
+    await page.mouse.move(100, 100);
+    await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+    await page.getByText("Change target").click();
+    await expect(buttonElement).toHaveText("Secondary target");
+    await buttonElement.hover();
+    await expect(getDataElementByValue(page, "tooltip")).toBeVisible();
+    await page.mouse.move(100, 100);
+    await expect(getDataElementByValue(page, "tooltip")).not.toBeVisible();
+  });
+
+  test(`should have the expected border radius styling`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<TooltipComponent />);
+
+    await expect(getDataElementByValue(page, "tooltip")).toHaveCSS(
+      "border-radius",
+      "4px"
+    );
+  });
+});
+
+test.describe("Accessibility tests for Tooltip component", () => {
+  test(`check accessibility tests for Default story`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Default />);
+
+    await checkAccessibility(page);
```

Comment:
As `click` is an action this can't be used in `expect` unfortunately. If we want to verify the tooltip is visible before running the AXE checks, we could amend this slightly like this. What do you think @stephenogorman?:

```suggestion
    await getDataElementByValue(page, "main-text").nth(0).click();
    
    const tooltip = page.getByRole("tooltip");
    await expect(tooltip).toVisible();
    await checkAccessibility(page);
```.

Reply:
Yes, that's a good idea. @divyajindel you could add this `await expect(page.getByRole("tooltip")).toVisible();` into each accessibility test before the `await checkAccessibility(page);` as James has suggested here.